### PR TITLE
HPCC-26529 elastic4hpcclogs automatically create index pattern

### PR DESCRIPTION
**Pull request by rpastrana**
- Creates index pattern upon podstart if it hasn't been created
- Mentions new behavior in README
- Updates chart version

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@xwang2713 please review

**Review by xwang2713 — COMMENTED** [review]
I don't see the detecting the first elastic pod code. With this code "hpcc-filebeat-pattern index pattern" will be created on all elastic pod, assume replicas > 1.

File: helm/managed/logging/elastic/values.yaml [hunk 1]
```diff
@@ -153,3 +153,38 @@ kibana:
...
+            ES_URL=http://localhost:9200
+            [ -z ${WAIT_KIBANA_READY_IN_SEC} ] && WAIT_KIBANA_READY_IN_SEC=120
+            END_TIME=$(expr $(date +%s) \+ ${WAIT_KIBANA_READY_IN_SEC})
+            while [[ "$(curl -s -o /dev/null -w '%{http_code}\n' $KIBANA_URL)" != "200" ]]; do
+              echo "\nKibana not ready..."
+              [ $(date +%s) -ge ${END_TIME} ] && break
+              sleep 1
+            done
+
+            httpcode=$(curl -s -o /dev/null -I -w "%{http_code}" -X GET "localhost:5601/api/index_patterns/index_pattern/hpcc-filebeat-pattern" -H 'kbn-xsrf: true')
```
Comment (rpastrana) [inline helm/managed/logging/elastic/values.yaml:177]: This should be using KIBANA_URL

**Comment by rpastrana** [comment]
> I don't see the detecting the first elastic pod code. With this code "hpcc-filebeat-pattern index pattern" will be created on all elastic pod, assume replicas > 1.

Good point. In this change, however, we're issuing commands directly to the Kibana service (not ES).

**Review by xwang2713 — COMMENTED** [review]
Code looks good according the Kibana README.md

**Comment by rpastrana** [comment]
@xwang2713 please review latest commit

**Review by xwang2713 — APPROVED** [review]
Looks good

**Pull request merged**